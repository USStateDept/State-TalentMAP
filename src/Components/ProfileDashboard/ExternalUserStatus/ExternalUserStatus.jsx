import React from 'react';
import PropTypes from 'prop-types';
import Avatar from '../../Avatar';
import Status from '../UserProfile/Status';
import USER_TYPES from '../../../Constants/UserTypes';
import MailToButton from '../../MailToButton';

const ExternalUserStatus = ({ initials, firstName, lastName, type, showMail, email }) => (
  <div className="usa-grid-full cdo-container">
    <div className="usa-grid-full cdo-container-inner section-padded-inner-container">
      <div className="profile-picture-container">
        <Avatar intiias={initials} firstName={firstName} lastName={lastName} />
        <div className="picture-status-container">
          <Status hideText />
        </div>
      </div>
      <div className="cdo-text-container">
        <div className="usa-grid-full">
          <div className="usa-width-one-whole cdo-header">
            {USER_TYPES[type]}
          </div>
          <div className="usa-width-one-whole cdo-name">
            <span className="cdo-name-text">{`${firstName} ${lastName}`}</span>
            { showMail ? <MailToButton email={email} /> : null}
          </div>
        </div>
      </div>
    </div>
  </div>
);

ExternalUserStatus.propTypes = {
  initials: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['cdo', 'ao', 'hr']).isRequired,
  showMail: PropTypes.bool,
  email: PropTypes.string,
};

ExternalUserStatus.defaultProps = {
  showMail: false,
  email: '',
};

export default ExternalUserStatus;
