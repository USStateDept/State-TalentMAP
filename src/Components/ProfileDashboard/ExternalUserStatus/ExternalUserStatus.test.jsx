import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ExternalUserStatus from './ExternalUserStatus';

describe('ExternalUserStatusComponent', () => {
  const props = {
    type: 'cdo',
    initials: 'JD',
    firstName: 'John',
    lastName: 'Doe',
  };

  it('is defined', () => {
    const wrapper = shallow(<ExternalUserStatus {...props} />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<ExternalUserStatus {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
