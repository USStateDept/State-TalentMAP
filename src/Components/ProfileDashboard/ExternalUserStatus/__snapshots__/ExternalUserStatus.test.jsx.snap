// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalUserStatusComponent matches snapshot 1`] = `
<div
  className="usa-grid-full cdo-container"
>
  <div
    className="usa-grid-full cdo-container-inner section-padded-inner-container"
  >
    <div
      className="profile-picture-container"
    >
      <Avatar
        className=""
        firstName="John"
        initials=""
        intiias="JD"
        lastName="Doe"
        onClick={[Function]}
      />
      <div
        className="picture-status-container"
      >
        <Status
          hideText={true}
        />
      </div>
    </div>
    <div
      className="cdo-text-container"
    >
      <div
        className="usa-grid-full"
      >
        <div
          className="usa-width-one-whole cdo-header"
        >
          Career Development Officer
        </div>
        <div
          className="usa-width-one-whole cdo-name"
        >
          <span
            className="cdo-name-text"
          >
            John Doe
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
